File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/util/Issue917Test.java
package org.babyfish.jimmer.sql.util;

import org.babyfish.jimmer.sql.model.Immutables;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class Issue917Test {

    @Test
    public void test() {
        Immutables.createTreeNode(draft -> {
            draft.setParentId(2L);
            Assertions.assertEquals(
                    "{\"parent\":{\"id\":2}}",
                    draft.toString()
            );
        });
    }
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/util/Issue917Test.kt
package org.babyfish.jimmer.sql.kt.util

import org.babyfish.jimmer.sql.kt.model.TreeNode
import org.babyfish.jimmer.sql.kt.model.classic.book.Book
import kotlin.test.Test
import kotlin.test.expect

class Issue917Test {

    @Test
    fun test() {
        TreeNode {
            parentId = 2L
            val text = this.toString()
            expect("""{"parent":{"id":2}}""") {
                text
            }
        }
    }
}